### PR TITLE
Client interface

### PR DESCRIPTION
**Pull request by barrucadu**
Make the Client an interface, not a struct. This allows overriding it with a no-op implementation if desired.

CC @WillSewell @zimbatm 

I think the lesson from all this is "don't submit a PR for a client library without actually trying to use it first."

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -112,119 +112,77 @@ type (
...
-	if tags == nil {
-		tags = map[string]string{
-			"cmd": os.Args[0],
-			"pid": strconv.Itoa(os.Getpid()),
-		}
-	}
-
-	c := Client{
+func NewClient(addr string, tags map[string]string) (Client, error) {
+	newConn, err := dial(addr, tags)
```
Comment (zimbatm) [inline client/client.go]: I believe this could happen only in the Run() function

File: client/interface.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package client
+
+import "github.com/pusher/stagger/conn"
+
+// Interface for Stagger clients.
```
Comment (zimbatm) [inline client/interface.go]: The Logger interface belongs in this file as well

**Comment by zimbatm** [comment]
If you rebase this on top of master I think it's good to go.

**Comment by barrucadu** [comment]
Rebased.

File: client/interface.go (hunk 2 above)
Comment (barrucadu) [inline client/interface.go]: Done

**Comment by zimbatm** [comment]
Thanks. And I'm going to try and fix the travis tests on another PR.

**Pull request merged**